job_templates/capsule_upgrade_-_ansible_default.erb

```diff
@@ -0,0 +1,58 @@
+<%#
+name: Capsule Upgrade Playbook
+snippet: false
+template_inputs:
+- name: target_version
+  required: true
+  input_type: user
+  advanced: false
+  value_type: plain
+  hidden_value: false
+- name: whitelist_options
+  required: false
+  input_type: user
+  advanced: false
+  value_type: plain
+  hidden_value: false
+model: JobTemplate
+job_category: Ansible Playbook
+description_format: "%{template_name}"
+provider_type: Ansible
+kind: job_template
+%>
+
+---
+- hosts: all
+  tasks:
+    - name: Gather the rpm package facts
+      package_facts:
+        manager: auto
+
+    - name: Fail if the target server is a Satellite server
+      fail:
+        msg: "This playbook cannot be executed on a Satellite server. Use only on a Capsule server."
+      when: "'satellite' in ansible_facts.packages"
+
+    - name: Install foreman-maintain if not present
+      package:
+        name: rubygem-foreman_maintain
+        state: present
+      when: "'rubygem-foreman_maintain' not in ansible_facts.packages"
+
+    - block:
+      - name: Upgrade Capsule server using foreman-maintain
+        command: foreman-maintain upgrade run --assumeyes --target-version=<%= input('target_version') %> --whitelist="<%= input('whitelist_options') %>"
+        register: result
+
+      - name: foreman-maintain upgrade return code is zero
+        debug:
+          msg: "Success! Capsule server upgrade completed. Current version of Capsule server server is {{ ansible_facts.packages['satellite-capsule'][0]['version'] }}."
```

Comment:
Is there an Ansible best practice to make sure that we are getting the newly updated version of the package and not the one that was gathered in facts before the play ran?

Reply:
It turns out the answer is yes, one option is https://docs.ansible.com/ansible/latest/modules/setup_module.html to re-gather (a subset of) facts

Reply:
great point, now I added the following to regather facts after upgrade.
```console
      - name: Re-Gather the rpm package facts after the upgrade
        package_facts:
          manager: auto
```